docs/adding_packages/folders_and_files.md

```diff
@@ -0,0 +1,192 @@
+can package that given folder.
+
+If it's no possible to maintain one recipe for all version, older version maybe moved to a separate folder.
+
+```yml
+versions:
+  "2.1.0":
+    folder: all
+  "2.0.0":
+    folder: all
+  "1.1.1":
+    folder: 1.x.x # Older version with different build system and options that are not compatible with newer version
+```
+
+### The _recipe folder_
+
+This contains every needed to build packages.
+
+#### `conandata.yml`
+
+This file lists **all the sources** that are needed to build the package. The most common examples are
+source code, build scripts, license files .
+
+The file is organized into two sections, `"sources"` and `"patches"`, each one of them contains the files that are required
+for each version of the library. Resources which need to be downloaded are listed under `"source"` should include a checksum
+to validate that they do not change. This helps to ensure the build are reproducible at a later point in time. Often
+modifications are requires for a variety of reasons, which ones are associated to which version are listed under the `"patches"`.
+
+```yml
+sources:
+  "9.0.0":
+    url: "https://github.com/fmtlib/fmt/archive/9.0.0.tar.gz"
+    sha256: "9a1e0e9e843a356d65c7604e2c8bf9402b50fe294c355de0095ebd42fb9bd2c5"
+```
+
+For more information about picking source tarballs, adding or removing versions, or what the rules are for patches, continue reading our
+[Sources and Patches](sources_and_patches.md) guide.
+
+> **Note**: Under our mission to ensure quality, patches undergo extra scrutiny. **Make sure to review** our
+> [modifying sources policy](sources_and_patches.md#policy-about-patching) before making changes.
+
+A detailed breakdown of all the fields can be found in [conandata_yml_format.md](conandata_yml_format.md). We **strongly** recommend adding the
+[patch fields](conandata_yml_format.md#patches-fields) to help track where patches come from and what issue they solve.
+
+Inside the `conanfile.py` recipe, this data is available in a `self.conan_data` attribute that can be used as follows:
+
+```py
+def source(self):
+    get(self, **self.conan_data["sources"][self.version], strip_root=True)
+```
+
+See the [Export Patches](sources_and_patches.md#exporting-patches) and [Applying Patches](sources_and_patches.md#applying-patches)
+for more use case and examples.
+
+#### `conanfile.py`
+
+This file is the recipe, it contains the logic to build the libraries from source for all the configurations.
+It's the single most important part of writing a package. Every `conanfile.py` should be accompanied by at least one
+[folder to test the generated packages](#test_package).
+
+Each recipe should derive the `ConanFile` class and implement key attributes and methods.
+
+* Basic attributes and conversions can be found in [`ConanFile` attributes](conanfile_attributes.md)
+* Some of the key methods are outline in this document and will link to more details
```

Comment:
```suggestion
* Some of the key methods are outlined in this document and will link to more details
```

Reply:
maybe also use present tense, as they already link (instead of `will link` in some unspecified future)